153_save_sensor_data/sensor_data_store.py

```diff
@@ -0,0 +1,248 @@
+# interval is reached.
+# @author Greg Ross, version 1, November 2,2021
+
+# Normal Packages
+import os
+import sys
+import csv
+import time
+from pathlib import Path
+# Special packages
+try:
+    import busio
+except RuntimeError:
+    print("Script failed during busio import. "
+          "Probably the sensor is not plugged in.")
+    sys.exit()
+    
+import adafruit_scd30
+
+# export statment is required to import board
+# Set the export statment command with a system call
+# in case it is not already set.
+# If this system call is not made, import board will fail
+# This system call sometimes works on my computer but usually fails
+os.system("export BLINKA_MCP2221=1")
+
+if 'BLINKA_MCP2221' not in os.environ:
+    print('Before running this program, type "export BLINKA_MCP2221=1"'
+          ' in the same terminal window.')
+    sys.exit()
+
+import board  # For MCP-2221
+
+
+
+# Variables to enhance precision of sensor
+
+# Temperature Offset - Ideally you should have a temperature sensor that is
+# independent of this sensor since this sensor generates heat as it is used
+# The temperature offset is the difference between ambient temperature and
+# the onboard temperature sensor. T_offset = Tscd_30 - Treferefence
+# set with scd.temperature_offset = OFFSET_VALUE. Every tick is 0.01*C (uint16)
+# Uses last value set before repowering.
+
+# There are also methods for calibration of the sensor but since they come 
+# calibrated it might not be good to change unless you have a sensor to compare
+# it against or have the right environment for autocalibration (requires access
+# to fresh air, and 5 days).
+
+# The measurement interval can be changed with this, but it still seems to take 
+# measurements at random time intervals just can be set "close" to this 
+# interval. The fastest interval combined with interpolation or just most recent
+# reading seems like the best way to me.
+# Can be increased to make a recent reading closer to another time step if 
+# we tried to synchronize it. Interpolation can be done either before data is 
+# saved or after data is retrieved from storage but prior to packaging in JSON.
+# scd.measurement_interval = INTERVAL 
+
+# Function Definitions
+
+def initialize_sensor():
+    ''' Function to initialize the sensor.
+        From Carter's Script at Adafruit, it is recommended to start with a 
+        frequency of 50 Khz due to "Clock Stretching". '''
+    i2c = busio.I2C(board.SCL, board.SDA, frequency=50000)
+    scd = adafruit_scd30.SCD30(i2c)
+    return scd
+ 
+def make_file_paths():
+    ''' Create the directories to store the saved data '''
+    Path("scd_data/raw").mkdir(parents=True,exist_ok=True)
+    Path("scd_data/interpolated").mkdir(exist_ok=True)
+
+
+def interpolate_linear(time_of_interest, data_set_low, data_set_high):
+    ''' # This function performs an interpolation to a time of interest
+        # given two data points outside of the time of interest using
+        # a linear method. '''
```

Comment:
```suggestion
    """
    Find the CO2, temperature, and humidity values between two data points.
    
    Given two data points and a time of interest between them,
    use linear interpolation to find the values at the given time
    and returns them in a dict.
    """
```
This is another example of multi-line docstring (maybe the phrasing could be better, but you get the idea).